pkg/propertyprovider/azure/provider.go

```diff
@@ -88,8 +90,64 @@ var _ propertyprovider.PropertyProvider = &PropertyProvider{}
 func (p *PropertyProvider) Start(ctx context.Context, config *rest.Config) error {
 	klog.V(2).Info("Starting Azure property provider")
 
+	podObj := client.Object(&corev1.Pod{})
 	mgr, err := ctrl.NewManager(config, ctrl.Options{
 		Scheme: scheme.Scheme,
+		Cache: cache.Options{
+			ByObject: map[client.Object]cache.ByObject{
+				podObj: {
+					// Set up field selectors so that API server will not send out watch events that
+					// are not relevant to the pod watcher. This is essentially a trade-off between
+					// in-memory check overhead and encoding/transmission overhead; for large clusters
+					// with frequent pod creation/deletion ops, the trade-off seems to be worth it based
+					// on current experimentation results.
+					Field: fields.AndSelectors(
+						fields.OneTermNotEqualSelector("spec.nodeName", ""),
+						fields.OneTermNotEqualSelector("status.phase", string(corev1.PodSucceeded)),
+						fields.OneTermNotEqualSelector("status.phase", string(corev1.PodFailed)),
+					),
+					// Drop irrelevant fields from the pod object; this can significantly reduce the
+					// CPU and memory usage of the pod watcher, as less data is stored in cache.
+					Transform: func(obj interface{}) (interface{}, error) {
```

Comment:
Where is this transform function run? I assume it's running on the pod watcher informer/indexer instead of the API server?